content/docs/render-props.md

```diff
@@ -0,0 +1,287 @@
+---
+id: render-props
+title: Render Props
+permalink: docs/render-props.html
+---
+
+The term ["render prop"](https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce) refers to a simple technique for sharing code between React components using a prop whose value is a function.
```

Comment:
It would be nice to kick off this page with a simple example and motivation to frame the conversation. The HOC doc starts with:

> Concretely, a higher-order component is a function that takes a component and returns a new component.
> `const EnhancedComponent = higherOrderComponent(WrappedComponent);`
> ...explain at high-level where and when HOCs are useful...

Paraphrasing the example code from @thejameskyle's #349, how about this: 

---

_The term ["render prop"](https://cdb.reacttraining.com/use-a-render-prop-50de598f11ce) refers to a simple technique for sharing code between React components using a prop whose value is a function._

A component with a render prop takes a function that returns a React element and calls it instead of implementing its own render logic.

```jsx
<DataProvider render={data => (
  <h1>Hello {data.target}</h1>
)}/>
```

By inverting control of the render method to the consumer of the component, render props provide a flexible way to encapsulate and share logic between components.

Libraries that use render props include [React Router](https://reacttraining.com/react-router/web/api/Route/Route-render-methods) and the enhanced input library [Downshift](https://github.com/paypal/downshift).

In this document, we’ll discuss why render props are useful, and how to write your own.